### PR TITLE
Wsr types icons

### PR DESCRIPTION
**Pull request by amitdahan**
### 🔦 Summary
Add Icons `.d.ts` generation

### CODE REVIEW

**Comment by giladsegal** [comment]
We need to make sure that a `.d.ts` file is being generated from all the named exports from icons root folder as well (or is this already happening?) 
@amitdahan

**Comment by amitdahan** [comment]
Mmmmm I don't understand if you described something else from what I did or not hhhhh
Call me when you got time and let's figure it out, but to me it sounds like it's already included in the PR :-D

File: scripts/proxy-wix-ui-icons/proxy-wix-ui-icons.js [hunk 1]
```diff
@@ -27,6 +29,8 @@ const proxyWixUiIconsCommon = (moduleName, iconsDir) => {
   const systemIconsFilesContentMap = prepareSystemIconsContentMap(moduleName);
   /* content of index.js file with named exports of all icons */
   const indexFileContent = prepareIndexFileContentForNamedExport(generalIconsFilesContentMap);
+  /* content of index.d.ts file with named exports of all icons */
+  const indexDefinitionFileCOntent = prepareIndexDefinitionFileContentForNamedExport(generalIconsFilesContentMap);
```
Comment (argshook) [inline scripts/proxy-wix-ui-icons/proxy-wix-ui-icons.js]:
```suggestion
  const indexDefinitionFileContent = prepareIndexDefinitionFileContentForNamedExport(generalIconsFilesContentMap);
```

on a side note, are functions names of this length really necessary? why not `createIndexFile` or something else shorter?

File: scripts/proxy-wix-ui-icons/proxy-wix-ui-icons.js [hunk 2]
```diff
@@ -35,6 +39,7 @@ const proxyWixUiIconsCommon = (moduleName, iconsDir) => {
   writeIconsFilesToFolderOnDisk(generalIconsFilesContentMap, iconsDir);
   writeIconsFilesToFolderOnDisk(systemIconsFilesContentMap, iconsDir, 'system/');
   writeIndexFileToDisk(indexFileContent, iconsDir);
+  writeIndexDefinitionFileToDisk(indexDefinitionFileCOntent, iconsDir);
```
Comment (argshook) [inline scripts/proxy-wix-ui-icons/proxy-wix-ui-icons.js]:
```suggestion
  writeIndexDefinitionFileToDisk(indexDefinitionFileContent, iconsDir);
```

File: scripts/proxy-wix-ui-icons/proxy-wix-ui-icons.js (hunk 1 above)
Comment (amitdahan) [reply scripts/proxy-wix-ui-icons/proxy-wix-ui-icons.js]: I basically mimicked existing code and added the word definition to wherever I needed to emit a `.d.ts`, I don't truly think it's necessary I just don't like steering away from a project's conventions, even with good intentions.

Comment (amitdahan) [reply scripts/proxy-wix-ui-icons/proxy-wix-ui-icons.js]: I can change it to whatever you like if you want :)

**Comment by amitdahan** [comment]
Removed root index.d.ts creation for `new-icons` per @shlomitc 's request

**Comment by simast** [comment]
I am closing this PR as importing icons from `new-icons/*` proxy has been deprecated and will be completely removed in next major version. The recommended way to import icons right now is to use `wix-ui-icons-common` package directly.

We should create a separate PR in `wix-ui` repo to expose TypeScript typings for icons there.

**Comment by simast** [comment]
Issue to track in `wix-ui` project: https://github.com/wix/wix-ui/issues/488

**Pull request closed without merging**